File: backend/app/models/post.py
from enum import StrEnum
from pydantic import BaseModel


class PostPlatform(StrEnum):
    instagram = "instagram"
    linkedin = "linkedin"


class PostOut(BaseModel):
    id: str
    club_id: str
    platform: PostPlatform
    url: str
    thumbnail_url: str | None
    is_active: bool
    created_at: str


class PostIn(BaseModel):
    platform: PostPlatform
    url: str
    thumbnail_url: str | None = None

File: mobile/types/post.ts
export type PostPlatform = 'instagram' | 'linkedin';

export interface Post {
  id: string;
  club_id: string;
  platform: PostPlatform;
  url: string;
  thumbnail_url: string | null;
  is_active: boolean;
  created_at: string;
}
